src/codecs/decoders.ts

```diff
@@ -0,0 +1,47 @@
+import * as wasmWebp from './webp/decoder';
+import * as browserWebp from './webp/decoder';
+
+import { createImageBitmapPolyfill, sniffMimeType } from '../lib/util';
+
+export interface Decoder {
+  name: string;
+  decode(blob: Blob): Promise<ImageBitmap>;
+  isSupported(): Promise<boolean>;
+  canHandleMimeType(mimeType: string): boolean;
+}
+
+// We load all decoders and filter out the unsupported ones.
+export const decodersPromise: Promise<Decoder[]> = Promise.all(
+  [
+    browserWebp,
+    wasmWebp,
+  ]
+  .map(async (decoder) => {
+    if (await decoder.isSupported()) {
+      return decoder;
+    }
+    return null;
+  }),
+// TypeScript is not smart enough to realized that I’m filtering all the falsy
+// values here.
+).then(list => list.filter(item => !!item)) as any as Promise<Decoder[]>;
+
+export async function findDecodersByMimeType(mimeType: string): Promise<Decoder[]> {
```

Comment:
This is exported, but I can't see it used anywhere.

Reply:
Yeah, exported it for completeness sake. Is exporting considered expensive?

Reply:
I think it prevents the name being mangled, but it also increases complexity. If I need to make changes to something that's exported, I need to consider its usage across the project. If it's scoped to a single module, I only need to look at that one file.

Same argument for public/private. If it weren't a burden, we'd export everything and make everything public.

Reply:
Mangling is a good argument. Will un-export

Reply:
This is still exported.